File: lemon/lemon_node/models/system/agent.js
/**
 *  Data model
 *  Agent 
 */

var mongoose	= require('mongoose');
var Schema		= mongoose.Schema;

var AgentSchema	= Schema({
	agent_id:	{ type: String, unique: true},
	name:		{ type: String, trim: true  },
	tags:	[ String ],
	entities:	[ {type: mongoose.Schema.Types.ObjectId, ref: 'inventory.Entity'} ]
});
AgentSchema.statics	= {
	load: function(agent_id, cb){
		this.findOne( {agent_id: agent_id})
			.populate('entities')
			.exec(cb);
	},
	listByTag: function(tag, cb){
		this.find( {tags: tag})
			.populate('entities')
			.sort('name')
			.exec(cb);
	},
	listExcludeTag: function( tag, cb){
		this.find( { tags: {$ne: tag} })
			.populate('entities')
			.exec(cb);
	},
	findByAgentId: function( agent_id, cb){
		this.findOne({agent_id: agent_id})
			.populate('entities')
			.exec(cb);
	}
};
AgentSchema.set('collection','agents');

module.exports	= function( __prefix__ ){	mongoose.model(__prefix__ + 'Agent',AgentSchema); };

File: lemon/lemonServer/models/Agent.py
'''
Created on 01 июля 2014 г.

@author: Andrean
'''
import models.base as base

class Agent(base.Base):
    '''
    Schema:
        agent_id:    String,
        name:        String,
        tags:     [  String  ],
        entities: [  dbref "Entities" ]
    '''    
    def __init__(self, dbref=None):
        self.__collection = "agents"
        self.__data = {
            'agent_id': None,
            'name':     "",
            'tags':     [],
            'entities': []
        }
        self.__dbref = dbref
        super().__init__(self.__dbref, self.__collection, self.__data)     